src/execution_impl.rs

```diff
@@ -84,3 +86,37 @@ pub fn run_function(
     let module = get_module(interface, bytecode)?;
     Ok(exec(limit, None, module, function, param)?.remaining_gas)
 }
+
+#[cfg(feature = "gas_calibration")]
```

Comment:
I think we can use the `exec` function above and have a function-like macro for the feature gas_calibration on the metering like in the other functions.

Reply:
Fixed